examples/platform/silabs/efr32/rs911x/hal/efx_spi.c

```diff
@@ -323,12 +298,12 @@ sl_status_t sl_wfx_host_post_bootloader_spi_transfer(void)
 sl_status_t sl_wfx_host_pre_lcd_spi_transfer(void)
 {
 #if SL_SPICTRL_MUX
-    if (sl_wfx_host_spi_cs_deassert() != SL_STATUS_OK)
-    {
-        return SL_STATUS_FAIL;
-    }
     xSemaphoreTake(spi_sem_sync_hdl, portMAX_DELAY);
 #endif // SL_SPICTRL_MUX
+    if (SL_STATUS_OK != sl_board_enable_display())
+    {
+       return SL_STATUS_FAIL;
```

Comment:
xSemaphoreTake is taken above. Make sure it is releases/given before return.

Reply:
returning SL_STATUS_FAIL for all the failures of sl_board_enable_display() function.
Possible to know is the actual return for failure case? or it will return always SL_STATUS_FAIL for all the failures.

Reply:
added the code to check status and releases semaphore if error occurs.